RuntimeDetour/Platforms/Runtime/DetourRuntimeNETCorePlatform.cs

```diff
@@ -5,20 +5,144 @@
+                    // it is the only method we have at the moment to detect it on non-Windows platforms
+                    isNet5Jit = typeof(object).Assembly.GetName().Version.Major >= 5;
+                }
+
+                try {
+                    getJit = clrjitPtr.GetFunction(nameof(getJit)).AsDelegate<d_getJit>();
+                } catch {
+                    DynDll.CloseLibrary(clrjitPtr);
+                    throw;
+                }
+            }
+
+            return getJit();
+        }
+
+        private static bool isNet5Jit;
+
+        protected static Guid GetJitGuid(IntPtr jit) {
+            int getVersionIdentIndex = isNet5Jit ? vtableIndex_ICorJitCompiler_getVersionIdentifier_net5
+                                                 : vtableIndex_ICorJitCompiler_getVersionIdentifier;
+            d_getVersionIdentifier getVersionIdentifier = ReadObjectVTable(jit, getVersionIdentIndex)
+                .AsDelegate<d_getVersionIdentifier>();
+            getVersionIdentifier(jit, out Guid guid);
+            return guid;
+        }
+
+        // The offset to use is determined in GetJitObject where other properties of the JIT are determined
+        private const int vtableIndex_ICorJitCompiler_getVersionIdentifier = 4;
+        private const int vtableIndex_ICorJitCompiler_getVersionIdentifier_net5 = 2;
+        [UnmanagedFunctionPointer(CallingConvention.ThisCall)]
+        private delegate void d_getVersionIdentifier(
+            IntPtr thisPtr, // ICorJitCompiler*
+            out Guid versionIdentifier
+            );
+
+        protected virtual int VTableIndex_ICorJitCompiler_compileMethod => 0;
+
+        protected static unsafe IntPtr* GetVTableEntry(IntPtr @object, int index)
+            => (*(IntPtr**) @object) + index;
+        protected static unsafe IntPtr ReadObjectVTable(IntPtr @object, int index)
+            => *GetVTableEntry(@object, index);
+
         protected override void DisableInlining(MethodBase method, RuntimeMethodHandle handle) {
             // https://github.com/dotnet/runtime/blob/89965be3ad2be404dc82bd9e688d5dd2a04bcb5f/src/coreclr/src/vm/method.hpp#L178
             // mdcNotInline = 0x2000
             // References to RuntimeMethodHandle (CORINFO_METHOD_HANDLE) pointing to MethodDesc
             // can be traced as far back as https://ntcore.com/files/netint_injection.htm
+        }
+
+        protected virtual void InstallJitHooks(IntPtr jitObject) => throw new PlatformNotSupportedException();
+
+        public override event OnMethodCompiledEvent OnMethodCompiled;
+
+        protected virtual void JitHookCore(
+                RuntimeTypeHandle declaringType,
+                RuntimeMethodHandle methodHandle,
+                IntPtr methodBodyStart, 
+                ulong methodBodySize,
+                RuntimeTypeHandle[] genericClassArguments,
+                RuntimeTypeHandle[] genericMethodArguments
+            ) {
```

Comment:
What happened to the spacing here?
```suggestion
        protected virtual void JitHookCore(
            RuntimeTypeHandle declaringType,
            RuntimeMethodHandle methodHandle,
            IntPtr methodBodyStart, 
            ulong methodBodySize,
            RuntimeTypeHandle[] genericClassArguments,
            RuntimeTypeHandle[] genericMethodArguments
        ) {
```

Reply:
I spaced it how I would with my normal `{` on new line style; it just doesn't really work here. I noticed that on the most recent pass over everything, but didn't touch it. I'll go ahead and fix it.